Extension/readme.developer.md

```diff
@@ -0,0 +1,358 @@
+# Developer Documentation
+
+## Table of Contents
+- [Developer Documentation](#developer-documentation)
+  - [Table of Contents](#table-of-contents)
+  - [Setup](#setup)
+    - [Required Tools](#required-tools)
+    - [Setting up the repository](#setting-up-the-repository)
+  - [Building the Extension](#building-the-extension)
+    - [When using F5 Debug](#when-using-f5-debug)
+    - [From Inside VSCode](#from-inside-vscode)
+    - [From Command line](#from-command-line)
+  - [Use of an isolated `vscode` environment](#use-of-an-isolated-vscode-environment)
+  - [Testing](#testing)
+    - [Unit tests](#unit-tests)
+    - [Scenario Tests](#scenario-tests)
+  - [Scripts](#scripts)
+    - [`yarn scripts`](#yarn-scripts)
+    - [`yarn show`](#yarn-show)
+    - [`yarn clean`](#yarn-clean)
+    - [`yarn test`](#yarn-test)
+    - [`yarn code`](#yarn-code)
+    - [`yarn generate-native-strings`](#yarn-generate-native-strings)
+    - [`yarn generate-options-schema`](#yarn-generate-options-schema)
+    - [`yarn copy-walkthrough-media`](#yarn-copy-walkthrough-media)
+    - [`yarn prep`](#yarn-prep)
+    - [`yarn import-edge-strings`](#yarn-import-edge-strings)
+    - [`yarn lint`](#yarn-lint)
+    - [`yarn compile`](#yarn-compile)
+    - [`yarn watch`](#yarn-watch)
+    - [`yarn verify`](#yarn-verify)
+    - [`yarn webpack`](#yarn-webpack)
+    - [`yarn translations-export`](#yarn-translations-export)
+    - [`yarn translations-generate`](#yarn-translations-generate)
+    - [`yarn translations-import`](#yarn-translations-import)
+    - [`yarn postinstall`](#yarn-postinstall)
+
+## Setup
+
+### Required Tools 
+
+* [Node.js](https://nodejs.org/en/download/) v16.*
+* Yarn - use `npm install -g yarn` to install
+
+### Setting up the repository
+
+`git clone https://github.com/microsoft/vscode-cpptools.git`
+
+`yarn install`
+
+## Building the Extension
+
+### When using F5 Debug 
+The `launch.json` entries now specify a `preLaunchTask` that will build the extension before launching using 
+the `yarn watch` command, and will wait until the build is ready. The watch command will continue to watch from 
+that point.
+
+If the extension is already built, the the watch will be very quick.
+
+### From Inside VSCode
```

Comment:
Could you normalize the references to `VS Code`? See for example https://code.visualstudio.com/docs/setup/setup-overview

Reply:
This wasn't fixed?

Reply:
That was not clear. I made them all 'VS Code' now